source/constructs/lib/agent/rename-resources/rename_resources.py

```diff
@@ -0,0 +1,185 @@
+                            headers=headers)
+    return response
+
+
+def on_create(event):
+    logger.info("Got create")
+
+
+def on_update(event):
+    logger.info("Got Update")
+    list_crawlers()
+
+
+def on_delete(event):
+    logger.info("Got Delete")
+
+
+def rename_database(new_database_name):
+    try:
+        glue_client.get_database(Name=new_database_name)
+        logger.info(f"The new database({new_database_name}) exists and does not need to be created")
+    except glue_client.exceptions.EntityNotFoundException as e:
+        logger.info(f"The new database({new_database_name}) does not exist and needs to be created")
+        response = glue_client.create_database(DatabaseInput={'Name': new_database_name})
+        # wait for database creation, several seconds
+        crawler_role_arn = f'arn:{partition}:iam::{account_id}:role/{solution_name}GlueDetectionJobRole-{region}'
+
+        # retry for grant permissions
+        num_retries = GRANT_PERMISSIONS_RETRIES
+        while num_retries > 0:
+            try:
+                lakeformation_client.grant_permissions(
+                    Principal={
+                        'DataLakePrincipalIdentifier': f"{crawler_role_arn}"
+                    },
+                    Resource={
+                        'Database': {
+                            'Name': new_database_name
+                        }
+                    },
+                    Permissions=['ALL'],
+                    PermissionsWithGrantOption=['ALL']
+                )
+            except Exception as e:
+                time.sleep(SLEEP_MIN_TIME)
+                num_retries -= 1
+            else:
+                break
+
+
+def rename_crawler(old_crawler_name, new_crawler_name, new_database_name, new_connection_name):
+    response = glue_client.get_crawler(Name=old_crawler_name)
+    logger.info(response)
+    old_crawler = response["Crawler"]
+    try:
+        glue_client.get_crawler(Name=new_crawler_name)
+        logger.info(f"The new crawler({new_crawler_name}) exists and does not need to be created")
+    except glue_client.exceptions.EntityNotFoundException as e:
+        logger.info(f"The new crawler({new_crawler_name}) does not exist and needs to be created")
+        jdbc_targets = old_crawler["Targets"]["JdbcTargets"]
+        if len(jdbc_targets) > 0:
+            logger.info("rename connection name")
+            for jdbc_target in jdbc_targets:
+                jdbc_target["ConnectionName"] = new_connection_name
+        glue_client.create_crawler(
+            Name=new_crawler_name,
+            Role=old_crawler["Role"],
+            DatabaseName=new_database_name,
+            Targets=old_crawler["Targets"],
+            Tags={
+                'AdminAccountId': admin_account_id,
+                'Owner': solution_name,
+            }
+        )
+
+
+def rename_connection(old_connection_name, new_connection_name):
+    response= glue_client.get_connection(Name=old_connection_name)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;old_connection_name&#39; (index: 0 | type: Unknown). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;new_connection_name&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness, emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles) using single or double underscore.

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).